File: SeongYeon/ts/6603.ts
// input
const input: string[] = require('fs')
  .readFileSync('test/input.txt')
  .toString()
  .trim()
  .split('\n');

// variable
let result = '';
let nums: number[] = [];
let k: number = 0;
let visited: boolean[] = [];

// solution
const dfs = (depth: number, start: number, str: string) => {
  if (depth === 6) {
    result += str + '\n';
    return;
  }

  for (let i = start; i < k; i++) {
    if (visited[i]) continue;
    visited[i] = true;
    dfs(depth + 1, i, str + nums[i] + ' ');
    visited[i] = false;
  }
};

input.forEach((element) => {
  if (element !== '0') {
    nums = element.split(' ').map((a) => +a);
    k = nums.shift()!;
    visited = [];
    dfs(0, 0, '');
    result += '\n';
  }
});

// output
console.log(result);

File: JiHwan/6603.py
def dfs(depth,start):
    if depth == 6:
        print(' '.join(map(str,answer)))

    for i in range(start,len(n)):
        answer.append(n[i])
        dfs(depth+1,i+1)
        answer.pop()



answer = []
while True:
    n = list(map(int,input().split()))
    if n[0] == 0:
        break
    del n[0]
    dfs(0,0)
    print()

